backend/catalogue/management/commands/retrieve_squidle_annotations_data.py

```diff
@@ -0,0 +1,211 @@
+
+class Command(BaseCommand):
+    network_boundary_layer: Layer
+    park_boundary_layer: Layer
+    http_session: requests.Session
+    api: SQAPI
+    additional_filters: list
+
+    def start_http_session(self) -> None:
+        retry_strategy = Retry(
+            total=3,
+            status_forcelist=[ 500, 502, 503, 504 ]
+        )
+        adapter = HTTPAdapter(max_retries=retry_strategy)
+        http_session = requests.Session()
+        http_session.mount("https://", adapter)
+        http_session.mount("http://", adapter)
+        self.http_session = http_session
+
+
+    def geojson_boundary(self, network: str, park: str = None) -> list:
+        boundary_layer = self.park_boundary_layer if park else self.network_boundary_layer
+        url = boundary_layer.server_url
+        params = {
+            'request':      'GetFeature',
+            'service':      'WFS',
+            'version':      '2.0.0',
+            'typeNames':    boundary_layer.layer_name,
+            'outputFormat': 'application/json',
+            'cql_filter': (f"RESNAME='{park}'" if park else f"NETNAME='{network}'")
+        }
+        r = self.http_session.get(url, params=params)
+        data = r.json()
+        boundary_type = data['features'][0]['geometry']['type']
+        if boundary_type == 'Polygon':
+            return [data['features'][0]['geometry']['coordinates']]
+        elif boundary_type == 'MultiPolygon':
+            return data['features'][0]['geometry']['coordinates']
+        else:
+            raise Exception(f"Unexpected boundary type: {boundary_type}")
+
+
+    def get_squidle_annotations(self, geojson: list, min: int, max: int, highlights: bool) -> str:
+        r = self.api.get("/api/annotation/tally/label")
+        r.template("models/annotation/tally_chart_mini.html")
+        r.results_per_page(10)
+
+        # filters
+        r.filter("point", "has", qf("media", "has", qf("poses", "any", qf("geom", "geo_in_mpolyh_xy", geojson))))
+        if highlights:
+            r.filter("point", "has", qf("media", "has", qf("annotations", "any", qf("annotations", "any", qf("tags", "any", qf("id", "eq", "348"))))))
+        if min:
+            r.filter("point", "has", qf("media", "has", qf("poses", "any", qf("dep", "gt", min))))
+        if max:
+            r.filter("point", "has", qf("media", "has", qf("poses", "any", qf("dep", "lte", max))))
+        
+        for additional_filter in self.additional_filters:
+            r._filters.append(additional_filter)
+
+        data = r.execute().text
+        tree = PyQuery(data)
+        if tree("div.tally-chart-row"):
+            return str(tree("div.tally-chart"))
+        else:
+            return None
+
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--skip_existing',
+            help='If true, skips retrieving squidle annotations data where we already have it',
+        )
+
+
+    def handle(self, *args, **options):
```

Comment:
General comment on this -- it's getting quite long-winded and nested, so I think at the very least a few comments about what it's doing, and the high-level steps along the way would be beneficial.

Reply:
Reworked it a bit to be a bit less confusing...